src/utils/list.c

```diff
@@ -16,6 +17,9 @@ void list_free(list *list) {
     now = list->first;
     while(now != NULL) {
         next = now->next;
+        if(list->free) {
```

Comment:
Should this also do the free on list_delete ?

Reply:
oh, yeah, didn't notice that existed.